pkg/provisioner/crdprovisioner.go

```diff
@@ -168,14 +168,30 @@ func (c *CrdProvisioner) CreateVolume(
 
 	azVolumeInstance, err := azV.Get(ctx, strings.ToLower(validVolumeName), metav1.GetOptions{})
 
-	if err == nil && azVolumeInstance.Status != nil && azVolumeInstance.Status.Error != nil {
-		// If the CreateVolume call is being retried after erroring out previously, set
-		// the status to nil to retrigger volume creation
-		azVolumeInstance.Status = nil
-		azVolumeUpdated, err := azV.UpdateStatus(ctx, azVolumeInstance, metav1.UpdateOptions{})
-		if err != nil || azVolumeUpdated == nil || azVolumeUpdated.Status != nil {
-			klog.Errorf("Failed to update azvolume resource for volume name (%s), error: %v", volumeName, err)
-			return nil, err
+	if err == nil {
+		if azVolumeInstance.Status != nil {
+			if azVolumeInstance.Status.Error != nil {
+				updatedInstance := azVolumeInstance.DeepCopy()
+				// If the CreateVolume call is being retried after erroring out previously, set
+				// the status to nil to retrigger volume creation
+				updatedInstance.Status = nil
+				// Updating the spec fields to keep it up to date with the request
+				updatedInstance.Spec.MaxMountReplicaCount = maxMountReplicaCount
+				updatedInstance.Spec.CapacityRange = capacityRange
+				updatedInstance.Spec.Parameters = parameters
+				updatedInstance.Spec.Secrets = secrets
+				updatedInstance.Spec.ContentVolumeSource = volumeContentSource
+				updatedInstance.Spec.AccessibilityRequirements = accessibilityReq
+				azVolumeUpdated, err := azV.UpdateStatus(ctx, updatedInstance, metav1.UpdateOptions{})
```

Comment:
Won't updateStatus only update the status? Should we also make azV.Update() call to update the spec?

Reply:
Addressed.